### ISSUE TITLE
teal buttons lack text contrast and fail WCAG

### ISSUE THREAD

**Issue by elioqoshi** [presentator/presentator#103]
The white text on the bright teal buttons lack contrast and fail any accessibility standards. 

WCAG 2.1 AA guidelines should be met at least

![image](https://user-images.githubusercontent.com/5436686/62450768-3df07d80-b76d-11e9-8565-ddc94015fb55.png)

**Comment by ganigeorgiev** [comment]
Changing to the closest AA text color doesn't look good in my opinion.
In v2.0.1 I've added a very light text-shadow to improve the readability as a compromise solution:
![text-shadow-screenshot](https://user-images.githubusercontent.com/8248071/62484004-da446f80-b7c1-11e9-8ebd-cd75f7a2af86.png)

If you think otherwise, please comment and I'll reopen the issue.

**Comment by dmartinjs** [comment]
I think this could help you
https://learnui.design/tools/accessible-color-generator.html?colors=38dcc9

<img width="220" alt="color-contrast" src="https://user-images.githubusercontent.com/53537199/62490434-905f8780-b7c9-11e9-9048-676bf96809a2.png">

**Comment by ganigeorgiev** [comment]
@dmartinjs Thanks for the link. I'll leave the issue open for discussion.

If anyone think that the success color should change, please add your suggestion as a comment and we'll decide what will stay based on the upvotes.

**Comment by elioqoshi** [comment]
It's not a matter of aesthetics when it comes to UI. Accessibility should be always priority. Blue and Orange have problems reaching WCAG while still looking good.

What @dmartinjs wrote is right. However buttons use 15px text size so 3:1 is not enough. 4:5 is required here which makes the nearest teal possible #008576

Alternatively, you can use Mineral UI colors which are built accessible from the ground up. 
https://mineral-ui.com/color

Please, let's not compromise accessibility for aesthetical reasons. Many governments even require WCAG AA to be reached to be used in their departments.

**Comment by ganigeorgiev** [comment]
I'm closing the issue, since is not very active.

I've replaced the larg green buttons on the landing page with white ones.
In the application itself I left them as they are because in my opinion they are readable enough and intuitive in the context they are placed.

If someone disagree, please comment and I'll reopen the issue.

**Comment by elioqoshi** [comment]
> In the application/admin panel/ I left them as they are because in my opinion they are readable enough and intuitive in the context they are placed.

It's not a matter of opinion but W3C standard of reaching WCAG AA or not. It can easily be checked with the Firefox Dev Tools Accessibility panel. If it fails WCAG, it should be changed. The context doesn't matter much here.

**Comment by ganigeorgiev** [comment]
@elioqoshi Although I don't fully agree with strictly implementing WCAG (in some cases it can produce even worst results) I understand why in general it is important and I'll try to improve the contrast ratio of the application buttons in the next release.

**Comment by elioqoshi** [comment]
> (in some cases it can produce even worst results)

Can you elaborate?

I just think that it's a low hanging fruit for things like buttons.

**Comment by ganigeorgiev** [comment]
@elioqoshi I guess probably depends on the screen, but I usually found the grey/color text contrasts to be more unreadable compared to the white text ones, although often the grey text has higher AA value (for example https://learnui.design/tools/accessible-color-generator.html?colors=30bdac).

Also keep in mind that text shadow, font style, antialiasing and other factors are usually not taken in consideration by the WCAG checkers.

Yes, generally speaking it is very easy to change, but I want to take my time experimenting with different color variations so I could keep the overall app style consistent (eg. not having the same ui control with white and black texts, different font-weight, etc.)

**Comment by ganigeorgiev** [comment]
@elioqoshi An update was released in v2.3.3:
![before_after](https://i.imgur.com/MktoMOJ.png)

It still doesn't pass the WCAG checkers, but once again they don't take the text-shadow and font style in consideration. Although there are others CVDs, here is a simulation with common color blindness variations (via https://www.color-blindness.com):

![color_blindness_test](https://i.imgur.com/VIQXVvi.png)

Once again, if there are actually people who have difficulties using Presentator with its current color scheme, please comment and I will reopen the issue.
